patches/tModLoader/Terraria/Main.cs.patch

```diff
@@ -3108,23 +3116,30 @@
 -					tServer.StartInfo.FileName = "TerrariaServer.exe";
 +					tServer.StartInfo.FileName = "tModLoaderServer.exe";
  #else
-+					if (Debugger.IsAttached) {
++					if (Debugger.IsAttached || File.Exists("Logs\\runtime")) 
++					{
 -					tServer.StartInfo.FileName = "TerrariaServer";
 +						tServer.StartInfo.FileName = "dotnet";
 +					}
 +					else {
-+						tServer.StartInfo.FileName = "NetFramework\\dotnet\\5.0.0\\dotnet";
++						// Dynamically read in the target runtime
++						using (StreamReader r = new StreamReader("tModLoader.runtimeconfig.json")) {
++							string json = r.ReadToEnd();
++							dynamic rConfigArr = JsonConvert.DeserializeObject(json);
++							string rVersion = rConfigArr.runtimeOptions.framework.version;
++							tServer.StartInfo.FileName = Path.Combine("dotnet", rVersion, "dotnet");
++						}
```

Comment:
Hmm. So, still gotta think on this; just posting prelim results in case a different idea.

Basically, that code works for creating shown console in Host&Play. 
`Process.GetCurrentProcess().MainModule.FileName` does not, even though you would expect it to. Still would be better to use.

The only difference is that the former is launching as a new window completely, the latter is not. Have to dig a bit deeper as to why it isn't working as desired.

Reply:
Windows was being stupid. Now uses the GetCurrProcess call correctly.
Shelling Path\dotnet.exe has it reuse the same console for the new process
Shelling Path\dotnet has it open a new console for the new process